pkg/imgpkg/internal/util/tag.go

```diff
@@ -29,3 +30,27 @@ func BuildDefaultUploadTagRef(item WithDigest, importRepo regname.Repository) (r
 	}
 	return uploadTagRef, nil
 }
+
+// BuildLegibleUploadTagRef Builds a legible tag from original image reference
+func BuildLegibleUploadTagRef(item WithDigest, itemRef string, importRepo regname.Repository) (regname.Tag, error) {
```

Comment:
I was looking at this code and I am not 100% sure this works in the scenario:
1. copy image from registry A to registry B
2. copy image from registry B to registry C

Looking at the parameters that are provided to this function I believe that `itemRef` on the second copy will be the reference from registry B. 

What we need is the original location of the Image. This might require more changes in the `imagedesc` package to save the original location of the Image as part of the ImageDescriptor type.

Can you validate if the case I pointed out in this comment is really happening?
Let me know if this comment is a bit to much information and might confuse you, I am free to jump on a Zoom call and look into this issue with you if it helps